iceoryx_posh/test/moduletests/test_version_info.cpp

```diff
@@ -0,0 +1,254 @@
+
+    VersionInfoSut(const Serialization& serial) noexcept
+        : VersionInfo(serial){};
+
+    uint16_t getVersionMajor()
+    {
+        return m_versionMajor;
+    }
+    uint16_t getVersionMinor()
+    {
+        return m_versionMinor;
+    }
+    uint16_t getVersionPatch()
+    {
+        return m_versionPatch;
+    }
+    uint16_t getVersionTweak()
+    {
+        return m_versionTweak;
+    }
+
+    VersionInfo::BuildDateStringType getBuildDateString()
+    {
+        return m_buildDateString;
+    }
+
+    VersionInfo::CommitIdStringType getCommitIdString()
+    {
+        return m_commitIdString;
+    }
+};
+} // namespace
+
+class VersionInfo_test : public Test
+{
+  public:
+    VersionInfo_test(){};
+    virtual ~VersionInfo_test(){};
+    virtual void SetUp(){};
+    virtual void TearDown(){};
+};
+
+
+TEST_F(VersionInfo_test, ValidMaxVersionTweak)
+{
+    VersionInfoSut versionInfoSut(65535u, 65535u, 65535u, 65535u, "abc", "efg");
+    EXPECT_EQ(versionInfoSut.getVersionMajor(), 65535u);
+    EXPECT_EQ(versionInfoSut.getVersionMinor(), 65535u);
+    EXPECT_EQ(versionInfoSut.getVersionPatch(), 65535u);
+    EXPECT_EQ(versionInfoSut.getVersionTweak(), 65535u);
+    EXPECT_EQ(std::string(versionInfoSut.getBuildDateString().c_str()), "abc");
+    EXPECT_EQ(std::string(versionInfoSut.getCommitIdString().c_str()), "efg");
+}
+
+TEST_F(VersionInfo_test, ValidMaxVersionFinalRelease)
+{
+    VersionInfoSut versionInfoSut(65535u, 65535u, 65535u, 0, "", "");
+    EXPECT_EQ(versionInfoSut.getVersionMajor(), 65535u);
+    EXPECT_EQ(versionInfoSut.getVersionMinor(), 65535u);
+    EXPECT_EQ(versionInfoSut.getVersionPatch(), 65535u);
+    EXPECT_EQ(versionInfoSut.getVersionTweak(), 0u);
+}
+
+TEST_F(VersionInfo_test, SerializationWorking)
+{
+    VersionInfoSut versionInfoSut1(65535u, 2u, 3u, 789u, "abc", "efg");
+    VersionInfoSut versionInfoSut2(versionInfoSut1.operator iox::cxx::Serialization());
+    EXPECT_EQ(versionInfoSut1.getVersionMajor(), versionInfoSut2.getVersionMajor());
+    EXPECT_EQ(versionInfoSut1.getVersionMinor(), versionInfoSut2.getVersionMinor());
+    EXPECT_EQ(versionInfoSut1.getVersionPatch(), versionInfoSut2.getVersionPatch());
+    EXPECT_EQ(versionInfoSut1.getVersionTweak(), versionInfoSut2.getVersionTweak());
+    EXPECT_EQ(std::string(versionInfoSut1.getBuildDateString().c_str()),
+              std::string(versionInfoSut2.getBuildDateString().c_str()));
+    EXPECT_EQ(std::string(versionInfoSut1.getCommitIdString().c_str()),
+              std::string(versionInfoSut2.getCommitIdString().c_str()));
+}
+
+TEST_F(VersionInfo_test, SerializationWorkingOnOurVersion)
+{
+    VersionInfoSut versionInfoSut1(VersionInfoSut::getCurrentVersion().operator iox::cxx::Serialization());
+    EXPECT_TRUE(versionInfoSut1 == VersionInfoSut::getCurrentVersion());
+}
+
+TEST_F(VersionInfo_test, ComparesWorkingForOurVersion)
+{
+    VersionInfoSut versionInfoSut1(VersionInfoSut::getCurrentVersion().operator iox::cxx::Serialization());
```

Comment:
Please use `static_cast<iox::cxx::Serialization>` instead of calling the conversion operator directly.